### ISSUE TITLE
Flickering When Rendering

### ISSUE THREAD

**Issue by petosiso** [mehdisadeghi/react-mathjax-preview#48]
Hi,

I am using react-mathjax-preview for rendering math-ml formulas. Everything works fine, but the initial rendering is flickering. For example, I am receiving this Math-ML formula:

`<math>
   <mtext>V =</mtext>
   <mfrac>
     <mtext>&#x3C0; . r . v</mtext>
     <mtext>3</mtext>
   </mfrac>
   <mtext/>
 </math>`

Then this string is used with mathjax-preview like this:
`<MathJax math={formula.Value} />`

At the beginning, everything is drawn in one line and no fraction is displayed. Then the number "3" moves to the denominator and the fraction is displayed. In last step size and padding is adjusted. This all takes about 0.5 - 1.0 second and the user sees the flickering and rearrangement of the entire form. 
Can you please help me with this issue?  Is it possible something like "generate formula in the background" and "then draw final formula to the screen"? Something like callback method, which will tell me that the formula is generated and should be displayed?

Thank you!

**Comment by mehdisadeghi** [comment]
Hi @petosiso,

Sorry for my late reply. I have noticed this delay myself but had no time to look at it. Do you experience the same on the MathJax website? Perhaps you can change the configuration according to the MathJax docs and see whether it helps.
Best of luck!

**Comment by petosiso** [comment]
Hi,

thank you for you reply. MathJax website (https://www.mathjax.org/#demo) works well without flickering or delay. What specific configuration do you have in mind and how to implement it? I am using react components.

Thanks!

**Comment by mehdisadeghi** [comment]
Perhaps [this one](https://github.com/mehdisadeghi/react-mathjax-preview/blob/master/index.jsx#L13)?

**Comment by svgpubs** [comment]
I am also struggling with the flickering display. It seems like the component is first rendering the text, and then transitioning into formatted equation. This is making my page jump quite a bit since there are multiple renders and changing equations. Can you make it only render only after the text is converted to format? I'd much rather have a delay (or css transition), than a jump between to states.

**Comment by svgpubs** [comment]
I suspect the source of the flickering is this pattern

```  
previewRef.current.innerHTML = sanitizedMath 
window.MathJax.Hub.Queue([
      'Typeset',
      window.MathJax.Hub,
      previewRef.current
  ])
```

The innerHTML is first being set to normal math text, and then the queque is processing it. The user is seeing all the steps. 

EDIT:
I just submitted a pull request.

**Comment by mehdisadeghi** [comment]
@svgpubs thanks for your PR. I just merged it!
